### PR TITLE
N°5547 - Object deletion fails if friendlyname too long

### PR DESCRIPTION
**Pull request by accognet**
Internal

### CODE REVIEW

**Review by Molkobain — CHANGES_REQUESTED** [review]
There is something wrong with your PR, 163 commits / 1600 files changed. It's most likely because you rebased on the wrong branch or forgot to change the target branch here.

image

File: tests/php-unit-tests/unitary-tests/core/DBObjectTest.php [hunk 1]
```diff
@@ -1235,4 +1235,52 @@ public function testConstructorMemoryFootprint():void
...
+				'😁😂🤣😃😄😅😆😗🥰😘😍😎😋😊😉😙😚',
+				'0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789-0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789-ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopq',
+			],
+		];
+	}
+
+	/**
+	 * N°5547 - Object deletion fails if friendlyname too long
+	 *
+	 * @covers       DBObject::DBIncrement
```
Comment (accognet) [inline tests/php-unit-tests/unitary-tests/core/DBObjectTest.php]:
```suggestion
	 * @covers       DBObject::DBDelete
```

**Comment by Molkobain** [comment]
Support PR review: Technically ok, ready for functional review

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 2]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+	 * Data provider for test deletion
+	 *  N°5547 - Object deletion fails if friendlyname too long
+	 *
+	 * @return array data
+	 */
+	public function testRecordObjDeletionProvider()
+	{
+		return [
+			'friendlyname longer than 255 characters ending with multi-bytes characters' => [
+				str_repeat('e',250),
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
				str_repeat('e', 250),
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 3]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+				'😁😂🤣😃😄😅😆😗🥰😘😍😎😋😊😉😙😚',
+			],
+			'friendlyname longer than 255 characters ending with single byte characters' => [
+				'😁😂🤣😃😄😅😆😗🥰😘😍😎😋😊😉😙😚',
+				str_repeat('e',250),
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
				str_repeat('e', 250),
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 4]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+		$oPerson = MetaModel::NewObject('Person', [
+			'first_name' => $sFirstName,
+			'name'       => $sName,
+			'org_id'     => 1,
+		]);
+		$oPerson->DBWrite();
+
+		$bDeletionOK = true;
+		try {
+			$oDeletionPlan = $this->InvokeNonPublicMethod(CMDBObject::class,'RecordObjDeletion',$oPerson, [$oPerson->GetKey()]);
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
			$oDeletionPlan = $this->InvokeNonPublicMethod(CMDBObject::class, 'RecordObjDeletion', $oPerson, [$oPerson->GetKey()]);
```

File: tests/php-unit-tests/unitary-tests/core/DBObjectTest.php [hunk 5]
```diff
@@ -1235,4 +1235,5 @@ public function testConstructorMemoryFootprint():void
 		$fTotalDuration = microtime(true) - $fStart;
 		echo 'Total duration: '.sprintf('%.3f s', $fTotalDuration)."\n\n";
 	}
+
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/DBObjectTest.php]:
```suggestion
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 6]
```diff
@@ -197,4 +246,5 @@ private function CreateUserForImpersonation($sLogin, $sProfileName, $sName, $sSu
 
 		return $oUser;
 	}
+
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
```

**Review by Molkobain — APPROVED** [review]

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 7]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+	/**
+	 * N°5547 - Object deletion fails if friendlyname too long
+	 *
+	 * @dataProvider testRecordObjDeletionProvider
+	 *
+	 */
+	public function testRecordObjDeletion( string $sFirstName, string $sName)
+	{
+		// Create a UserRequest with 2 contacts
```
Comment (piRGoif) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]: This comment is false and can be removed

File: tests/php-unit-tests/unitary-tests/core/DBObjectTest.php [hunk 8]
```diff
@@ -1199,7 +1199,7 @@ public function GetObjectReloadCount($sClass, $sKey)
 	{
 		return $this->aReloadCount[$sClass][$sKey] ?? 0;
 	}
-	
+
```
Comment (piRGoif) [inline tests/php-unit-tests/unitary-tests/core/DBObjectTest.php]: Maybe revert this change if possible ?

**Review by piRGoif — CHANGES_REQUESTED** [review]
Thanks for the modifications ! Good work !
I just see those 2 details remaining

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 9]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+	/**
+	 * Data provider for test deletion
+	 *  N°5547 - Object deletion fails if friendlyname too long
+	 *
+	 * @return array data
+	 */
+	public function testRecordObjDeletionProvider()
+	{
+		return [
+			'friendlyname longer than 255 characters ending with multi-bytes characters' => [
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
			'friendlyname longer than 255 characters which will be truncated on a multi-bytes characters' => [
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 10]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+				str_repeat('e', 250),
+				'😁😂🤣😃😄😅😆😗🥰😘😍😎😋😊😉😙😚',
+			],
+			'friendlyname longer than 255 characters ending with single byte characters' => [
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
			'friendlyname longer than 255 characters which will be truncated after a single byte characters' => [
```

**Review by Molkobain — APPROVED** [review]

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php (hunk 7 above)
Comment (accognet) [reply tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]: done

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 11]
```diff
@@ -168,6 +170,52 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
 		CMDBObject::SetTrackInfo($sInitialTrackInfo);
 	}
 
+	/**
+	 * Data provider for test deletion
+	 *  N°5547 - Object deletion fails if friendlyname too long
+	 *
+	 * @return array data
+	 */
+	public function testRecordObjDeletionProvider()
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
	public function RecordObjDeletionProvider()
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 12]
```diff
@@ -168,6 +170,52 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+				'😁😂🤣😃😄😅😆😗🥰😘😍😎😋😊😉😙😚',
+				str_repeat('e', 250),
+			],
+		];
+	}
+
+	/**
+	 * N°5547 - Object deletion fails if friendlyname too long
+	 *
+	 * @dataProvider testRecordObjDeletionProvider
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
	 * @dataProvider RecordObjDeletionProvider
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 13]
```diff
@@ -168,6 +170,52 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+		$oPerson->DBWrite();
+
+		$bDeletionOK = true;
+		try {
+			$oDeletionPlan = $this->InvokeNonPublicMethod(CMDBObject::class, 'RecordObjDeletion', $oPerson, [$oPerson->GetKey()]);
+		}
+		catch (CoreException $e) {
+			$bDeletionOK = false;
+		}
+		$this->assertTrue($bDeletionOK);
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php:216]:
```suggestion
		// We d'ont need to test the result (truncated string), it's already done in \DBObject::SetTrim() with N°3448
		$this->assertTrue($bDeletionOK);
```

File: tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php [hunk 14]
```diff
@@ -168,6 +170,53 @@ public function testCurrentChangeUnderImpersonation($sTrackInfo=null, $sExpected
...
+		$oPerson->DBWrite();
+
+		$bDeletionOK = true;
+		try {
+			$oDeletionPlan = $this->InvokeNonPublicMethod(CMDBObject::class, 'RecordObjDeletion', $oPerson, [$oPerson->GetKey()]);
+		}
+		catch (CoreException $e) {
+			$bDeletionOK = false;
+		}
+		// We d'ont need to test the result (truncated string), it's already done in \DBObject::SetTrim() with N°3448
```
Comment (Molkobain) [inline tests/php-unit-tests/unitary-tests/core/CMDBObjectTest.php]:
```suggestion
		// We don't need to test the result (truncated string), it's already done in \DBObject::SetTrim() with N°3448
```

**Comment by Molkobain** [comment]
Accepted during support PRs review with Anne-Cath, Pierre & Romain.
PR will reviewed in functional review next week.

We would like it to be merged in support/3.1 as the previous / similar issue was also merged in 31.

**Comment by Molkobain** [comment]
Accepted during functional review
@accognet you can "squash & merge", please mind to keep the PR number in the commit.

**Comment by Molkobain** [comment]
Also mind to rebase your PR branch on support/3.1 AND to change the target branch here on the PR to support/3.1

**Pull request merged**